### ISSUE TITLE
Wallet process isn't killed when Brave wallet is disabled in extensions until browser restart

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#6237]
## Description 

Wallet process isn't killed when Brave wallet is disabled in extensions

## Steps to Reproduce

   1. Click on Crypto Wallet to start the wallet process
   2. Disable `Brave Wallet` under extension
   3. Open a new tab and visit brave://wallet, shows error page
   4. Open browser task manager, Wallet process is still running even though its disabled under extension

## Actual result:

![image](https://user-images.githubusercontent.com/17010094/65910002-cec38e00-e397-11e9-8dba-a50cb8ebd135.png)

## Expected result:
Should kill the process if the extension is disabled

## Reproduces how often: 

Easy

## Brave version (brave://version info)

Brave | 0.69.130 Chromium: 77.0.3865.90 (Official Build) (64-bit)
-- | --
Revision | 58c425ba843df2918d9d4b409331972646c393dd-refs/branch-heads/3865@{#830}
OS | Linux

## Version/Channel Information:

- Can you reproduce this issue with the current release? No
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the dev channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc: @ryanml @bbondy

**Comment by bbondy** [comment]
future restarts don't show it though, so probably not a big deal that we sit on this one for a bit.

**Comment by kjozwiak** [comment]
@bbondy assuming this is the same as per our earlier conversation regarding MM detecting Crypto Wallets even after disabling which causes https://github.com/brave/brave-browser/issues/6181? Double checking before creating a new issue.

**Comment by bbondy** [comment]
This won't be needed anymore when this is implemented:
https://github.com/brave/brave-browser/issues/7503

**Comment by NumDeP** [comment]
Hi @srirambv I had an issue with updating the browser on openSuse which has now been resolved thanks to francois but after I was given the commands to essentially refresh the repo and sort of re-install the browser, I noticed that the crypto walltet came on as well. I could easily turn it off and it would remain off unlike that issue over a year ago where it was practically hard coded and wouldn't turn off but is this expected or is it a bug? I pinned it for you as shown in the attachment.

![Screenshot_20220428_062130](https://user-images.githubusercontent.com/31957540/165682609-2feeada9-6df2-4fcc-9503-08eb82755759.png)

One other thing, it's clearly stated that crypto wallets is now deprecated, does that mean the team isn't expected to work on that particular product any longer and in a future update would it effectively be removed from the browser?
